### PR TITLE
allow blank passwords

### PR DESCRIPTION
**Pull request by aochsner**
**Affects:** \<sonar-collector-version-number>.

---

### CODE REVIEW

File: src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java [hunk 1]
```diff
@@ -447,7 +447,7 @@ private HttpHeaders createHeaders(String token) {
 
     protected HttpHeaders setHeaders(RestUserInfo userInfo){
         if(Objects.isNull(userInfo)) return null;
-        if(StringUtils.isNotBlank(userInfo.getUserId())&& StringUtils.isNotBlank(userInfo.getPassCode())){
+        if(StringUtils.isNotBlank(userInfo.getUserId())){
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java:450]: If I recollect this was intentional, let me recollect the reason behind this change and get back to you.

**Review by rvema — CHANGES_REQUESTED** [review]
under review, let me get back to you on this

File: src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java (hunk 1 above)
Comment (aochsner) [reply src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java:450]: Sure, and let me provide my context (I actually still need to test this on my local sonarqube).  Our SonarQube requires authentication, so I have to either provide a username/password or an API key.  I'd rather provide an API key.  However, this logic gets in the way https://github.com/Hygieia/hygieia-codequality-sonar-collector/blob/master/src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java#L107-L111 

The `URL_RESOURCES_AUTHENTICATED` URI requires 'System Administrator' permission which our service account accessing SonarQube won't get.  But we are able to get it via the `URL_RESOURCES` URI.  I'm not sure why there are 2 ways of doing the same thing TBH.  Anyways, if I use a username/password then it works.  If I use API_TOKEN, it doesn't.

Basically all I'm trying to do is circumvent that logic (while keeping it in place for whoever put it in and still needs it)

Comment (aochsner) [reply src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java:449]: Tested on my local w/ the latest changes I just pushed

Comment (rvema) [reply src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java:449]: the changes look fine

**Review by rvema — CHANGES_REQUESTED** [review]
can you increment the application version to 3.2.1-SNAPSHOT so that this change can be merged

**Comment by aochsner** [comment]
Done!

**Review by rvema — APPROVED** [review]
lgtm
